### PR TITLE
[test] add allure annotation

### PR DESCRIPTION
**Pull request by troferoller**

### CODE REVIEW

**Comment by troferoller** [comment]
reworked sometests

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 1]
```diff
@@ -152,6 +168,13 @@ test('Registering domain', async () => {
 })
 
 test('When querying for not existing domain, returns FindError', async () => {
```
Comment (AlexStroke) [inline packages/client/test/integration/test-node/test/main.spec.ts:170]:
```suggestion
test('Query not existing domain', async () => {
```

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 2]
```diff
@@ -152,6 +168,13 @@ test('Registering domain', async () => {
...
+  await allure.owner("dulger");
+  await allure.label("feature", "Domains");
+  await allure.label("permission", "no_permission_required");
+  await allure.label("sdk", "Java Script");
+  await allure.label("sdk_test_id", "query_all_domains");
```
Comment (AlexStroke) [inline packages/client/test/integration/test-node/test/main.spec.ts:175]:
```suggestion
  await allure.label("sdk_test_id", "query_not_existing_domain");
```

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 3]
```diff
@@ -217,6 +240,13 @@ describe('Events API', () => {
 
 describe('Setting configuration', () => {
```
Comment (AlexStroke) [inline packages/client/test/integration/test-node/test/main.spec.ts:241]: This test needs to be deleted. We are testing this in Torii api tests.

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 4]
```diff
@@ -231,6 +261,13 @@ describe('Setting configuration', () => {
 
 describe('Blocks Stream API', () => {
```
Comment (AlexStroke) [inline packages/client/test/integration/test-node/test/main.spec.ts:262]: This test is useful, but let's put it in a separate file and don't mark it with allure for now, because this test tests the API, not the SDK.

**Review by AlexStroke — CHANGES_REQUESTED** [review]

File: packages/client/test/integration/test-node/test/compatibility.spec.ts [hunk 5]
```diff
@@ -18,8 +18,12 @@ setupPeerTestsLifecycle()
 describe('Compatibility Matrix tests', () => {
   // Based on https://github.com/AlexStroke/iroha-java/blob/007a9ac55991cd8a2b519e62a10144156d9f8301/modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/InstructionsTest.kt#L134
   test('register domain', async (ctx) => {
-    await allure.feature(ctx, 'Domains')
-    await allure.story(ctx, 'Account registers a domain')
+    await allure.owner(ctx, "dulger");
+    await allure.label(ctx, "feature", "Domains");
+    await allure.label(ctx, "permission", "no_permission_required");
+    await allure.label(ctx, "sdk", "Java Script");
+    await allure.label(ctx, "sdk_test_id", "register_domain");
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility.spec.ts:25]: Set up beforeEach for: await allure.label("permission", "no_permission_required"); 
                                       await allure.label("sdk", "Java Script");
                                       await allure.owner(ctx, "dulger");

File: packages/client/test/integration/test-node/test/main.spec.ts (hunk 4 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/main.spec.ts:262]: Move to the other file to create a special test file for API

File: packages/client/test/integration/test-node/test/main.spec.ts (hunk 3 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/main.spec.ts:241]: Deleted

File: allure-report_2024-07-15_17-23-10/data/categories.json
Comment (AlexStroke) [inline allure-report_2024-07-15_17-23-10/data/categories.json:1]: We always should `git ignore` generated allure files.

**Review by AlexStroke — CHANGES_REQUESTED** [review]

**Review by AlexStroke — APPROVED** [review]

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 6]
```diff
@@ -82,6 +91,8 @@ test('AddAccount instruction with name length more than limit is not committed',
 })
 
 test('Ensure properly handling of Fixed type - adding Fixed asset and querying for it later', async () => {
```
Comment (AlexStroke) [inline packages/client/test/integration/test-node/test/main.spec.ts:93]:
```suggestion
test(Mint fixed asset', async () => {
  await allure.label(ctx, "feature", "Asset");
  await allure.label(ctx, "sdk_test_id", "mint_fixed_asset");
  await allure.label(ctx, "story", "Mint a fixed asset");
  
```

**Review by AlexStroke — CHANGES_REQUESTED** [review]

**Review by AlexStroke — CHANGES_REQUESTED** [review]

File: package.json [hunk 7]
```diff
@@ -46,5 +46,10 @@
     "vite": "^5.1.5",
     "vitest": "^1.3.1",
     "zx": "^7.2.0"
+  },
+  "dependencies": {
+    "@iroha2/client": "workspace:^",
+    "@iroha2/crypto-target-web": "workspace:^",
+    "@iroha2/data-model": "workspace:^"
```
Comment (0x009922) [inline package.json:53]: Remove

File: .npmrc [hunk 8]
```diff
@@ -1 +1,3 @@
 publish-branch=iroha2
+# FILE: .npmrc
+@iroha2:registry=https://nexus.iroha.tech/repository/npm-group/
```
Comment (0x009922) [inline .npmrc:3]: This line is for those who is _using_ SDK's published packages. Id doesn't make sense to be here, so, remove.

File: packages/client/test/api.spec.ts
Comment (0x009922) [inline packages/client/test/api.spec.ts:1]: Why did you remove it from the integration tests? Please move it back.

File: packages/client/test/integration/test-node/test/compatibility.spec.ts [hunk 9]
```diff
@@ -5,53 +5,66 @@
...
-        ),
-      )
-    )
-      .as('Ok')
-      .batch.enum.as('Identifiable')
-      .enum.as('Domain')
+    await allure.step(ctx,"Result of operation matches the expected result", async()=>{
+      result!.as('Ok')
+      result!.batch?.enum?.as('Identifiable')
+      result!.enum?.as('Domain')
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility.spec.ts:64]:
```suggestion
      result!.as('Ok').batch.enum.as('Identifiable').enum.as('Domain')
```

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 10]
```diff
@@ -1,87 +1,116 @@
...
 
-          pipe(
-            sugar.identifiable.newAssetDefinition(id, datamodel.AssetValueType('BigQuantity')),
-            sugar.instruction.register,
-            sugar.executable.instructions,
-          ),
-        )
-      }),
-    )
+  await allure.step(ctx, "Creating asset with long asset name", async()=>{
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/main.spec.ts:31]: This step is meaningless. It is a plain JS operation and is not related to allure & compat matrix.

File: packages/client/test/integration/test-node/test/main.spec.ts [hunk 11]
```diff
@@ -1,87 +1,116 @@
...
-      pre,
-      pipe(
-        [normal, incorrect].map((id) => pipe(sugar.identifiable.newAccount(id, []), sugar.instruction.register)),
-        sugar.executable.instructions,
-      ),
-    )
+  await allure.step(ctx, "Create accounts with normal length name and long length name", async() => {
+    normal = sugar.accountId('bob', 'wonderland')
+    incorrect = sugar.accountId('0'.repeat(2 ** 14), 'wonderland')
+  })
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/main.spec.ts:83]: Same - meaningless to wrap into a `step`.

**Review by 0x009922 — CHANGES_REQUESTED** [review]
- Format code with `pnpm -w format:fix`
- Preferrably if you only add additions to `compatibility.spec.ts` and keep everything related to Allure there. I understand that some tests from `main.spec.ts` are ideologically useful for compat matrix - so, just move them into the `compatibility.spec.ts`

**Review by 0x009922 — CHANGES_REQUESTED** [review]
It may be not in my scope, but the use of `step`s is confusing in current implementation. Could you explain the rationale behind it?

File: packages/client/test/integration/test-node/test/compatibility/api.spec.ts
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/api.spec.ts:1]: we are going to add `listenings` into the matrix

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+    expect(existingDefinitions).toContainEqual(normalAssetDefinitionId)
+    expect(existingDefinitions).not.toContainEqual(invalidAssetDefinitionId!)
+  })
+})
+
+test('Register an account with long account name', async (ctx) => {
+  await allure.allureId(ctx, '4077')
+  await allure.label(ctx, 'feature', 'Account')
+  await allure.label(ctx, 'sdk_test_id', 'register_account_with_long_account_name')
+  await allure.label(ctx, 'story', 'Register an account with long name')
```
[the remark this answers is not in the document]
Comment (alexstroke1) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:129]: I don't really understand the value of this suggestion. All annotations that we can put in advance, we put before each. If we can't annotate in advance, it's better to mark up each specific test with the necessary information for readability.

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+      pre,
+      pipe(
+        sugar.assetId(sugar.accountId('alice', 'wonderland'), sugar.assetDefinitionId('XOR', 'wonderland')),
+        sugar.find.assetById,
+      ),
+    )
+  })
+  await allure.step(ctx, 'Result when exist', async () => {
+    expect(result!.tag === 'Err').toBe(true)
+  })
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: Deleted step, and make for all result part

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+
+  let tooLongAssetName
+  let invalidAssetDefinitionId
+
+  await allure.step(ctx, 'Account registers an asset with long asset name', async () => {
+    tooLongAssetName = '0'.repeat(2 ** 14)
+    invalidAssetDefinitionId = sugar.assetDefinitionId(tooLongAssetName, 'wonderland')
+  })
+
+  await allure.step(ctx, 'The asset with long name is rejected', async () => {
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: renamed The normal asset is registered

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+
+  const normalAssetDefinitionId = sugar.assetDefinitionId('xor', 'wonderland')
+
+  let tooLongAssetName
+  let invalidAssetDefinitionId
+
+  await allure.step(ctx, 'Account registers an asset with long asset name', async () => {
+    tooLongAssetName = '0'.repeat(2 ** 14)
+    invalidAssetDefinitionId = sugar.assetDefinitionId(tooLongAssetName, 'wonderland')
+  })
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: renamed Given an asset with long asset name

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 16]
```diff
@@ -0,0 +1,229 @@
...
+      expect(final).toEqual(DOMAIN_NAME)
+    })
+  })
+})
+
+beforeEach(async (ctx) => {
+  await allure.owner(ctx, 'dulger')
+  await allure.label(ctx, 'permission', 'no_permission_required')
+  await allure.label(ctx, 'sdk', 'Java Script')
+})
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: deleted repetition

[the remark this answers is not in the document]
Comment (0x009922) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: But I still see it there

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (0x009922) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:129]: It is called __domain-specific language__.

`allure` utility is general, not specific to your domain. It doesn't know that you are using labels `feature`, `sdk_test_id`, `story`. It doesn't know that `allureId` is supposed to be a number, so that you have to pass it as a string instead.

Here I see it useful to define a small domain-specific function that actually _defines_ your testing domain, known attributes, __statically__. The value is that you are less likely to make a mistake/typo, you can have a hint from type system, you repeat less. Instead of writing time and time again this:

```ts
await allure.allureId(ctx, '4077')  
await allure.label(ctx, 'feature', 'Account')  
await allure.label(ctx, 'sdk_test_id', 'register_account_with_long_account_name')
await allure.label(ctx, 'story', 'Register an account with long name')
```

You can write this:

```ts
await annotate(ctx, {
    // expects a number
    allureId: 4077,

    feature: 'Account',
    stdTestId: 'register_account_with_long_account_name',
    story: 'Register an account with long name',
    
    // Ooops, typo! Get a hint and a type error
    feeture: 'Bad'
})
// internally, this function will call `allure` for you
```

And using this function that _knows_ your domain is a more robust approach. In addition, it's just less boilerplate and is arguably more expressive, as it doesn't contain extra noise.

I wouldn't say that reducing code repetition and adding layers of abstraction is always a good idea, but in this case, it is clearly (for me) worth it.

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+
+async function assignAllureAttributes(
+    ctx: TaskContext,
+    attrs: { id: string, feature: string, sdkTestId: string, story: string }
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:24]:
```suggestion
    attrs: { id: number, feature: string, sdkTestId: string, story: string }
```

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+import { datamodel, sugar } from '@iroha2/data-model'
+import { CLIENT_CONFIG } from '@iroha2/test-configuration'
+import { Seq } from 'immutable'
+
+setupPeerTestsLifecycle()
+
+beforeEach(async (ctx) => {
+  await allure.owner(ctx, 'dulger')
+  await allure.label(ctx, 'permission', 'no_permission_required')
+  await allure.label(ctx, 'sdk', 'Java Script')
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:19]: Consider removing these attributes from `beforeEach` and putting them into `assignAllureAttributes` - so that they are all defined in a single place.

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+})
+
+async function assignAllureAttributes(
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:22]:
```suggestion
async function annotateAllure(
```

Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 18 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 17 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 20]
```diff
@@ -0,0 +1,242 @@
...
+import { pipe } from 'fp-ts/function'
+import { describe, expect, TaskContext, test } from 'vitest'
+import { clientFactory, setupPeerTestsLifecycle } from '../util'
+import { datamodel, sugar } from '@iroha2/data-model'
+import { CLIENT_CONFIG } from '@iroha2/test-configuration'
+import { Seq } from 'immutable'
+
+setupPeerTestsLifecycle()
+
+async function annotateAllure(
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:16]: You can also move it to shared utils and re-use it in all other files.

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 21]
```diff
@@ -0,0 +1,242 @@
...
+      sdkTestId: 'register_domain',
+    })
+
+    const DOMAIN_NAME = 'new_domain_name'
+    let result
+    let final
+
+    const { pre, client, getBlocksListener } = clientFactory()
+    const blocks = await getBlocksListener()
+    await allure.step(ctx, "Account registers a new domain with name 'new_domain_name'", async () => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:45]:
```suggestion
    await allure.step(ctx, "Submit transaction to register a new domain", async () => {
```

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 22]
```diff
@@ -0,0 +1,242 @@
...
+      await blocks.wait(async () => {
+        await client.submitExecutable(
+          pre,
+          pipe(sugar.identifiable.newDomain(DOMAIN_NAME), sugar.instruction.register, sugar.executable.instructions),
+        )
+      })
+    })
+    await allure.step(ctx, "Creating 'result' variable wich consist DomainId of created domain", async () => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:53]:
```suggestion
    await allure.step(ctx, "Ensure domain is registered with `FindDomainById` query", async () => {
```

I would merge this step with the next one (with assertions)

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 23]
```diff
@@ -0,0 +1,242 @@
...
+    await allure.step(ctx, 'The domain is registered', async () => {
+      result!.as('Ok').batch.enum.as('Identifiable').enum.as('Domain')
+
+      final = result!.__c[0].batch.enum.__c[0].enum.__c[0].id.name
+      expect(final).toEqual(DOMAIN_NAME)
+    })
+  })
+})
+
+test('Register asset with too long type', async (ctx) => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:77]:
```suggestion
test('Register asset with a too long name', async (ctx) => {
```

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 24]
```diff
@@ -0,0 +1,242 @@
...
+
+  await allure.step(ctx, 'Given an assent with normal name', async() =>{
+    normalAssetDefinitionId = sugar.assetDefinitionId('xor', 'wonderland')
+  })
+
+
+  await allure.step(ctx, 'Given an asset with long asset name', async () => {
+    tooLongAssetName = '0'.repeat(2 ** 14)
+    invalidAssetDefinitionId = sugar.assetDefinitionId(tooLongAssetName, 'wonderland')
+  })
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:100]: Meaningless to wrap into steps

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 25]
```diff
@@ -0,0 +1,242 @@
...
+
+  await allure.step(ctx, 'Account register transaction with 2 instructions', async () => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:102]:
```suggestion
  await allure.step(ctx, 'Submit a transaction registering 2 assets (valid and invalid)', async () => {
```

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 26]
```diff
@@ -0,0 +1,242 @@
...
+              sugar.identifiable.newAssetDefinition(id, datamodel.AssetValueType('BigQuantity')),
+              sugar.instruction.register,
+              sugar.executable.instructions,
+            ),
+          )
+        }),
+      )
+    })
+  })
+  await allure.step(ctx, 'The normal asset is registered', async () => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:121]:
```suggestion
  await allure.step(ctx, 'Ensure that only the valid one is registered', async () => {
```

**Review by 0x009922 — CHANGES_REQUESTED** [review]

**Review by 0x009922 — CHANGES_REQUESTED** [review]
Please apply changes to other steps in the same manner.

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 26 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:121]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 25 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:102]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 20 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:16]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 21 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:45]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 22 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:53]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 24 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:100]: step deleted

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 27]
```diff
@@ -0,0 +1,242 @@
...
+    ctx: TaskContext,
+    attrs: { id: number, feature: string, sdkTestId: string, story: string }
+) {
+  await allure.label(ctx, 'allureId', String(attrs.id))
+  await allure.label(ctx, 'feature', attrs.feature)
+  await allure.label(ctx, 'sdk_test_id', attrs.sdkTestId)
+  await allure.label(ctx, 'story', attrs.story)
+  await allure.owner(ctx, 'dulger')
+  await allure.label(ctx, 'permission', 'no_permission_required')
+  await allure.label(ctx, 'sdk', 'Java Script')
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: ready

File: packages/client/test/integration/test-node/test/util.ts [hunk 28]
```diff
@@ -38,10 +49,13 @@ export function setupPeerTestsLifecycle() {
...
   })
 
-  beforeEach(async () => {
+  beforeEach(async (ctx) => {
     await TestPeer.clearPeerStorage()
     startedPeer = await TestPeer.startPeer()
     await waitForGenesisCommitted(clientFactory().pre)
+    await allure.owner(ctx, 'dulger')
+    await allure.label(ctx, 'permission', 'no_permission_required')
+    await allure.label(ctx, 'sdk', 'Java Script')
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/util.ts:58]: ready

File: packages/client/test/integration/test-node/test/util.ts [hunk 29]
```diff
@@ -17,6 +18,16 @@ export function freeOnTestFinished<T extends Free>(object: T): T {
  * Sets up Vitest hooks to run a clean Iroha peer
  * for each test.
  */
+export async function annotateAllure(
```
[the remark this answers is not in the document]
Comment (troferoller) [reply packages/client/test/integration/test-node/test/util.ts:21]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 23 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 30]
```diff
@@ -0,0 +1,226 @@
...
+
+  // Assert
+  const asset = Seq(result.as('Ok').batch.enum.as('Vec'))
+    .map((x) => x.enum.as('Identifiable').enum.as('Asset'))
+    .find((x) => x.id.definition_id.name === ASSET_DEFINITION_ID.name)
+
+  expect(asset?.value).toEqual(datamodel.AssetValue('Fixed', datamodel.FixedPointI64(DECIMAL)))
+})
+
+test('Query not existing domain', async (ctx) => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:201]: Still has inconsistent steps

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 31]
```diff
@@ -0,0 +1,226 @@
...
+      .as('Ok')
+      .batch.enum.as('Vec')
+      .map((val) => val.enum.as('Identifiable').enum.as('Account').id)
+
+    expect(existingAccounts).toContainEqual(normal!)
+    expect(existingAccounts).not.toContainEqual(incorrect!)
+  })
+})
+
+test('Mint fixed asset', async (ctx) => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:159]: Doesn't have steps at all

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 32]
```diff
@@ -0,0 +1,226 @@
...
+  })
+
+  const { client, pre, getBlocksListener } = clientFactory()
+  const blocks = await getBlocksListener()
+
+  let normal
+  let incorrect
+
+  normal = sugar.accountId('bob', 'wonderland')
+  incorrect = sugar.accountId('0'.repeat(2 ** 14), 'wonderland')
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:133]: Why separating variable declaration from value assignments?

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 33]
```diff
@@ -0,0 +1,226 @@
...
+    const blocks = await getBlocksListener()
+    await allure.step(ctx, "Submit transaction to register a new domain", async () => {
+      await blocks.wait(async () => {
+        await client.submitExecutable(
+          pre,
+          pipe(sugar.identifiable.newDomain(DOMAIN_NAME), sugar.instruction.register, sugar.executable.instructions),
+        )
+      })
+    })
+    await allure.step(ctx, "Query domain by `FindDomainById`", async () => {
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:42]: Querying step is still split from assertion for no good reason.

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts [hunk 34]
```diff
@@ -0,0 +1,226 @@
...
+    await annotateAllure(ctx, {
+      id: 4075,
+      feature: 'Domain',
+      story: 'Account registers a domain',
+      sdkTestId: 'register_domain',
+    })
+
+    const DOMAIN_NAME = 'new_domain_name'
+    let result
+    let final
```
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:30]: What is the purpose of defining `final` here?

Define it right where you need it unless you don't need to move a value from a nested scope; and use `const` if you aren't going to re-assign.

And please read this article very carefully: http://dmitrysoshnikov.com/ecmascript/javascript-the-core-2nd-edition-rus/

File: packages/client/test/integration/test-node/test/compatibility/multisignature.spec.ts
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/multisignature.spec.ts:1]: Please do not include this test suite in this PR.

File: packages/client/test/integration/test-node/test/compatibility/api.spec.ts
Comment (0x009922) [inline packages/client/test/integration/test-node/test/compatibility/api.spec.ts:1]: Please do not include this test suite in this PR.

File: packages/client/test/integration/test-node/test/util.ts
Comment (0x009922) [inline packages/client/test/integration/test-node/test/util.ts:1]: Restore this file to the original.

**Review by 0x009922 — CHANGES_REQUESTED** [review]

File: packages/client/test/integration/test-node/test/compatibility/api.spec.ts
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/api.spec.ts:1]: ready

File: packages/client/test/integration/test-node/test/compatibility/multisignature.spec.ts
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/multisignature.spec.ts:1]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 33 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:42]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 32 above)
Comment (0x009922) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:133]: Haven't seen the commits yet

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 34 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:30]: ready

File: packages/client/test/integration/test-node/test/util.ts
Comment (troferoller) [reply packages/client/test/integration/test-node/test/util.ts:1]: ready

File: packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts (hunk 30 above)
Comment (troferoller) [reply packages/client/test/integration/test-node/test/compatibility/compatibility.spec.ts:201]: deleted

**Review by 0x009922 — APPROVED** [review]
